### ISSUE TITLE
[red-knot] support Literal[None] as a spelling for None

### ISSUE THREAD

**Issue by carljm** [astral-sh/ruff#14981]
Per the spec, `Literal[None]` is a valid spelling for `None` (that is, in red-knot terms, `Type::Instance(types.NoneType)`.) And e.g. `Literal[1, None]` is an alternate spelling for `Literal[1] | None`. We don't support this, but we should.

**Comment by Glyphack** [comment]
I tried reproducing this using the following test but it passes:

```python
x1: Literal[1, None]
def f():
  # revealed: Literal[1] | None
  reveal_type(x1)

x2: Literal[None]
def f2():
  # revealed: None
  reveal_type(x2)
```

Did I miss something in the description?

**Comment by cake-monotone** [comment]
On a slightly different note, there’s also the issue of class literals.
For example, `Literal[str, int]` should be represented as `type[str]` | `type[int]`.

**Comment by carljm** [comment]
Sorry, this was my mistake, I made an assumption and didn't check it!

> `Literal[str, int]` should be represented as `type[str]` | `type[int]`

Yeah, perhaps. The problem with this is that `Literal[str]` and `type[str]` are different types (the latter includes all subclasses, the former does not) and can behave differently, so it could be confusing to display them the same. On the other hand, it can be confusing to display types that aren't part of the spec / we don't accept if spelled by the user in an annotation. (We will have this latter problem also for intersections.) I don't have a clear plan in mind yet for how we should handle this :/

One other note on `Literal` is that we currently split different "kinds" of literal, so we show `Literal[1, 2] | Literal["foo"]`, whereas the spec (and every other type checker) prefer `Literal[1, 2, "foo"]`. It's probably better for us to switch to the latter, since there's no ambiguity.

**Comment by carljm** [comment]
https://github.com/astral-sh/ruff/issues/14988
